pkg/controller/authentication.go

```diff
@@ -129,10 +130,21 @@ func (c *Controller) loginEndpoint(ctx iris.Context) {
 			return
 		}
 
+		ctx.SetCookie(&http.Cookie{
+			Name:     "M-Token",
+			Value:    token,
+			Path:     "/",
+			Domain:   c.configuration.APIDomainName,
+			Expires:  time.Now().Add(7 * 24 * time.Hour),
+			MaxAge:   0,
+			Secure:   true,
+			HttpOnly: true,
+			SameSite: http.SameSiteStrictMode,
+		})
```

Comment:
This is duplicated in several spots. It should probably be lifted into its own function and allow for other parameters like expiration, same site and secure be configurable.

HttpOnly should be a general requirement, the UI does not _need_ to know the actual token. 

But other things like secure or domain name should definitely be configurable for self-hosted deployments.